### PR TITLE
Adding some flag ranges for the validators

### PR DESCRIPTION
**Pull request by aoliver44**
Added a bunch of flag options for the validator function. These are the ones that i think would be most helpful, but definitely can update in future.

### CODE REVIEW

**Comment by aoliver44** [comment]
I dont know why its failing its test. There were some helpful error messages, which I fixed with the additional commits. But now all I'm seeing is "Error: Process completed with exit code 1." I'm not sure what to do with that off the top of my head.

**Comment by aoliver44** [comment]
I think it is probably due to the quit() code you have in line 301 of the test_options.R file. When I run that myself, I get an error "Error: cannot change value of locked binding for 'quit'". 

Is this what you expect? Likewise, when i run the file in Rstudio, I get a interactive that asks me if i want to quit my session ("y/n/c"). So it seems like quit() is not being overwritten?

Ok, thats about as much guessing as I can do.

File: lib/options.R [hunk 1]
```diff
@@ -59,7 +60,18 @@ argument_groups <- list(
 # this method should cat an error and quit() when they are not satisfied
 # or cat a warning message
 validators <- list(
-  cor_level=validate_numeric(min=0, max=1, min_warning=list(.6, "correlation below this level makes no sense"))
+  cor_level=validate_numeric(min=0, max=1, min_warning=list(0.6, "correlation below 0.6 is departing from the spirit of this competition - to group things that likely contain redundant information")),
+  k_splits=validate_numeric(min=2, max_warning=list(6, "these are a lot of splits...using this many splits with small data is probably unwise")),
+  prevalence=validate_numeric(min=0, max=1),
+  abundance=validate_numeric(min=0),
+  lowest_level=validate_numeric(min=1, min_warning=list(2, "values below 2 may include an artificial taxonomic root")),
+  max_level=validate_numeric(min=1, max=1000, max_warning=list(16, "you have many hierarchial levels, which may increase run time")),
```
Comment (mmmckay) [inline lib/options.R:68]:
```suggestion
  max_level=validate_numeric(min=1, max=1000, max_warning=list(16, "you have many hierarchical levels, which may increase run time")),
```

File: lib/options.R [hunk 2]
```diff
@@ -59,7 +60,18 @@ argument_groups <- list(
 # this method should cat an error and quit() when they are not satisfied
 # or cat a warning message
 validators <- list(
-  cor_level=validate_numeric(min=0, max=1, min_warning=list(.6, "correlation below this level makes no sense"))
+  cor_level=validate_numeric(min=0, max=1, min_warning=list(0.6, "correlation below 0.6 is departing from the spirit of this competition - to group things that likely contain redundant information")),
+  k_splits=validate_numeric(min=2, max_warning=list(6, "these are a lot of splits...using this many splits with small data is probably unwise")),
+  prevalence=validate_numeric(min=0, max=1),
+  abundance=validate_numeric(min=0),
+  lowest_level=validate_numeric(min=1, min_warning=list(2, "values below 2 may include an artificial taxonomic root")),
```
Comment (mmmckay) [inline lib/options.R:67]: maybe in the future we could have the tree offset by one so this makes more sense to a user that doesn't know the model

**Comment by mmmckay** [comment]
> I think it is probably due to the quit() code you have in line 301 of the test_options.R file. When I run that myself, I get an error "Error: cannot change value of locked binding for 'quit'".
> 
> Is this what you expect? Likewise, when i run the file in Rstudio, I get a interactive that asks me if i want to quit my session ("y/n/c"). So it seems like quit() is not being overwritten?
> 
> Ok, thats about as much guessing as I can do.

I think this is happening because you didn't add test values in for the new validations, and I have some example value that is triggering quit. My sample value for lowest_level is 0 [here](https://github.com/aoliver44/taxaHFE/blob/development/tests/test_options.R#L184)

**Comment by aoliver44** [comment]
Awesome, great catch. Yeah I probably did think it was a >= type situation. Lowest level was the one flag that tripped me up more. We set it to 3 by default, which means L2 (phylum) are the lowest level. Kind of an artificial choice there are worth making that clear in README documentation, which i still have to get around to.

**Pull request merged**